File: client/src/features/employees/schemas/dutyEnum.ts
export const dutyValues = [
  "TEACHER",
  "ADM",
  "THERAPIST",
  "MENTOR",
  "SYSTEM",
] as const;

export type Duty = typeof dutyValues[number];

export const dutyLabels = {
  TEACHER: "Professor(a)",
  ADM: "Administrativo",
  THERAPIST: "Terapeuta",
  MENTOR: "Mentor(a)",
  SYSTEM: "Sistema",
} as const satisfies Record<Duty, string>;

File: server/src/main/java/aprimorar/pessoas/colaborador/api/contract/DutyEnum.java
package aprimorar.pessoas.colaborador.api.contract;

import io.swagger.v3.oas.annotations.media.Schema;

@Schema(description = "Função/Papel do funcionário")
public enum DutyEnum {
    TEACHER("Professor"),
    ADM("Administrativo"),
    THERAPIST("Terapeuta"),
    MENTOR("Mentor"),
    SYSTEM("Sistema");

    private final String description;

    DutyEnum(String description) {
        this.description = description;
    }

    public String getDescription() {
        return description;
    }
}
